pinot-query-runtime/src/main/java/org/apache/pinot/query/runtime/operator/window/value/LagValueWindowFunction.java

```diff
@@ -78,7 +78,9 @@ public List<Object> processRows(List<Object[]> rows) {
     int numRows = rows.size();
     Object[] result = new Object[numRows];
     if (_defaultValue != null) {
-      Arrays.fill(result, 0, _offset, _defaultValue);
+      // We only fill up to numRows to handle the case where the offset
+      // is larger than numRows
```

Comment:
This comment is a little confusing, isn't `numRows` the total size of the result array?

Reply:
ah you are right. Lookin at the line right below it, we proceed from _offset to numRows to fill the rest of the values